src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolderDescriptor.java

```diff
@@ -75,8 +100,74 @@ public String getCategoryId() {
         return r;
     }
 
+    /**
+     * Gets the {@link FolderIconDescriptor}s applicable for this folder type.
+     *
+     * @since FIXME
+     */
+    public List<FolderIconDescriptor> getIconDescriptors() {
+        List<FolderIconDescriptor> r = new ArrayList<FolderIconDescriptor>();
+        for (FolderIconDescriptor p : FolderIconDescriptor.all()) {
+            if (p.isApplicable(clazz.asSubclass(AbstractFolder.class))) {
+                r.add(p);
+            }
+        }
+        StaplerRequest request = Stapler.getCurrentRequest();
+        if (request != null) {
+            AbstractFolder folder = request.findAncestorObject(AbstractFolder.class);
+            if (folder != null) {
+                return DescriptorVisibilityFilter.apply(folder, r);
+            }
+        }
+        return r;
+    }
+
+    // TODO remove once baseline 2.0
+    public String getDescription() {
+        Stapler stapler = Stapler.getCurrent();
+        if (stapler != null) {
+            try {
+                WebApp webapp = WebApp.getCurrent();
+                MetaClass meta = webapp.getMetaClass(this);
+                Script s = meta.loadTearOff(JellyClassTearOff.class).findScript("newInstanceDetail");
```

Comment:
This method was required in order to not break descriptions in the case where somebody upgrades folders where the plugin that extends folders has not been updated. 

It is an edge case for sure... you need to have upgraded to Jenkins 2.0 without updating the plugins that extend `Folder`... or you need to have a plugin that has not been updated yet... say https://github.com/jenkinsci/literate-plugin/blob/master/src/main/resources/org/cloudbees/literate/jenkins/LiterateMultibranchProject/newInstanceDetail.jelly which is maybe 45 installations or so... but that is just one example where this is required (and yes I'll probably be updating literate with all the scm api changes and will fix that issue... but until that time this method allows the correct description for literate to be displayed when using the latest folders plugin)